creator/studies/migrations/0031_patch_dataservice.py

```diff
@@ -0,0 +1,91 @@
+# Generated by Django 2.2.24 on 2021-11-02 18:23
+
+"""
+This migration patches the new attributes into the DataService Study
+entities using the spreadsheet found in this migrations folder.
+"""
+
+from django.conf import settings
+from django.db import migrations
+
+import logging
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+def patch_helper(kf_id, study_dict):
+    """
+    Helper function to patch the data in study_dict to the kf_id Study in
+    DataService.
+    """
+    # Patch DataService
+    try:
+        resp = requests.patch(
+            f"{settings.DATASERVICE_URL}/studies/{kf_id}",
+            json=study_dict,
+            timeout=settings.REQUESTS_TIMEOUT,
+            headers=settings.REQUESTS_HEADERS,
+        )
+    except requests.exceptions.RequestException as e:
+        raise Exception(f"Problem updating study {kf_id}: {e}")
+    
+    # Study not found in DataService
+    if resp.status_code == 404:
+        logger.warning(
+            f"Study {kf_id} was not found in the DataService when trying to "
+            "update its short_code, domain, and program fields."
+        )
+        return
+    # Raise an error if it looks like study failed to update
+    if not resp.status_code == 200 or "results" not in resp.json():
+        error = resp.json()
+        if "_status" in error:
+            error = error["_status"]
+        if "message" in error:
+            error = error["message"]
+        raise Exception(f"Problem updating study {kf_id}: {error}")
+
+
+def patch_attrs(apps, schema_editor):
+    """
+    Patch the short_code, domain, and program attributes of Study objects
+    in the DataService using the provided spreadsheet.
+    """
+    Study = apps.get_model("studies", "Study")
+    for study in Study.objects.all():
+        kf_id = study.kf_id
+        study_dict = {
+            "short_code": study.short_code,
+            "domain": study.domain,
+            "program": study.program,
+        }
+        patch_helper(kf_id, study_dict)
+
```

Comment:
One last request! Can you log the study that was updated along with the study_dict? I'm testing it and I like the warning logs for the ones not found. Thought it would be nice to see what got updated too